### ISSUE TITLE
The encoding (CMap) specified by a font is missing for copied PDF pages from this specific PDF

### ISSUE THREAD

**Issue by DanielJackson-Oslo** [Hopding/pdf-lib#135]
When merging this specific PDF using the copy page methods discussed in #134, it ends up garbling all text, and giving the following error code in Adobe Reader: "The encoding (CMap) specified by a font is missing." 

**This happens both in the branch 1.0.0 and the 0.x.x branch.** 

The input PDF before copying passes qpdf --check with no errors, and is this file:
[a6d6d730-a2ec-11e9-9500-7b856429f3db.pdf](https://github.com/Hopding/pdf-lib/files/3376537/a6d6d730-a2ec-11e9-9500-7b856429f3db.pdf)

The output PDF also passes qpdf, but contains no fonts for the pages from the input PDF:
[2019-01-30-Tenåsen Regnskap-1644.00-NOK.pdf](https://github.com/Hopding/pdf-lib/files/3376540/2019-01-30-Tenasen.Regnskap-1644.00-NOK.pdf)

Interestingly enough the output PDF is fully readable in Google Chrome, but broken in Adobe Reader and Mac OS Preview.

Might be fixed for my use case with #121 , but unsure what's going on.

**Comment by Hopding** [comment]
Hello @DanielJackson-Oslo!

I just released a fix for this to the v1.0.0 beta branch (it's the latest `@beta` tag). You can install it with `npm`:
```
npm install --save pdf-lib@1.0.0-beta.2
```
or yarn:
```
yarn add pdf-lib@1.0.0-beta.2
```
It's also available on the unpkg CDN:
* https://unpkg.com/pdf-lib@1.0.0-beta.2/dist/pdf-lib.js
* https://unpkg.com/pdf-lib@1.0.0-beta.2/dist/pdf-lib.min.js

The issue was a slight oversight in the decode function for PDFNames. Only the `0-9` hex digits were being decoded. The digits `A-F` were not being decoded. The fix was just to add these digits to the regex: https://github.com/Hopding/pdf-lib/commit/d123b59e72b6c63567887b70934f5720d249938a.

I'll release a fix to the non-beta version (`0.6.4`) as well.

**Comment by Hopding** [comment]
I just released the fix to `0.6.4-rc2`. You can install it with `npm`:
```
npm install --save pdf-lib@0.6.4-rc2
```
or yarn:
```
yarn add pdf-lib@0.6.4-rc2
```
It's also available on the unpkg CDN:
* https://unpkg.com/pdf-lib@0.6.4-rc2/dist/pdf-lib.js
* https://unpkg.com/pdf-lib@0.6.4-rc2/dist/pdf-lib.min.js

**Comment by DanielJackson-Oslo** [comment]
You're the best @Hopding . Will test as soon as I get a chance to.

**Comment by DanielJackson-Oslo** [comment]
@Hopding The pdf-lib@1.0.0-beta.2 build gives the following error:

```Error: Cannot find module '@pdf-lib/fontkit'```

The build works fine in -beta.1. I've flushed my node_modules folder and package-lock.json and tried it again, with no luck.

I cannot for the life of me understand how that error can come from this https://github.com/Hopding/pdf-lib/commit/d123b59e72b6c63567887b70934f5720d249938a though...

**Comment by Hopding** [comment]
Are you embedding custom fonts? If so, you have to install Fontkit separately in v1.0.0: https://github.com/Hopding/pdf-lib/tree/Rewrite#fontkit-installation.

**Comment by DanielJackson-Oslo** [comment]
Ah, the problem is that for some reason pdf-lib@1.0.0-beta.2 does not in fact load pdf-lib@1.0.0-beta.2. The fix in d123b59 is not in the code in node_modules, so god knows which version it actually is.

The sha is: `sha512-godVXwb4UIVBn7Z/ZGCPgOupyJgW6L+LRsysxbPP8eE2CQZ8y1xfNHWHa8nIkJACg0SgB5Y7d2336GmHw3mEDA==`

Do you get the fix with  pdf-lib@1.0.0-beta.2 on your end?

My package.json looks like this:
`		"pdf-lib": "^1.0.0-beta.2",`

and my package-lock like this:

```json

		"pdf-lib": {
			"version": "1.0.0-beta.2",
			"resolved": "https://registry.npmjs.org/pdf-lib/-/pdf-lib-1.0.0-beta.2.tgz",
			"integrity": "sha512-godVXwb4UIVBn7Z/ZGCPgOupyJgW6L+LRsysxbPP8eE2CQZ8y1xfNHWHa8nIkJACg0SgB5Y7d2336GmHw3mEDA==",
			"requires": {
				"@pdf-lib/standard-fonts": "^0.0.4",
				"pako": "^1.0.10",
				"png-ts": "^0.0.3",
				"tslib": "^1.10.0"
			},
			"dependencies": {
				"tslib": {
					"version": "1.10.0",
					"resolved": "https://registry.npmjs.org/tslib/-/tslib-1.10.0.tgz",
					"integrity": "sha512-qOebF53frne81cf0S9B41ByenJ3/IuH8yJKngAX35CmiZySA0khhkovshKK+jGCaMnVomla7gVlIcc3EvKPbTQ=="
				}
			}
		},
```

**Comment by DanielJackson-Oslo** [comment]
(Not embedding any fonts)

**Comment by DanielJackson-Oslo** [comment]
Cloning and building the Rewrite branch works, and with it the fix also works 🎉  However, installing from npm does not work but gives a broken branch like described above :/

**Comment by Hopding** [comment]
Looks like I forgot a `prepublish` step for the `1.0.0-beta.2` release. I just fixed this and released `1.0.0-beta.3`. Please let me know if this one works!

**Comment by DanielJackson-Oslo** [comment]
This one works! Thank you @Hopding . Closing the issue. 🎉

**Comment by Hopding** [comment]
I just released version `0.6.4`. The full release notes are available [here](https://github.com/Hopding/pdf-lib/releases/tag/v0.6.4).

You can install it with `npm`:
```
npm install --save pdf-lib@0.6.4
```
or yarn:
```
yarn add pdf-lib@0.6.4
```
It's also available on the unpkg CDN:
* https://unpkg.com/pdf-lib@0.6.4/dist/pdf-lib.js
* https://unpkg.com/pdf-lib@0.6.4/dist/pdf-lib.min.js
